src/Molecules/Duplicable/Duplicable.jsx

```diff
@@ -1,60 +1,61 @@
 import React, { useState } from 'react'
-
-const computeNextKey = duplicated => {
-  let nextKey = 0
-
-  duplicated.map(duplicate => {
-    const key = parseInt(duplicate.key)
-
-    if ('key' in duplicate && key >= nextKey) {
-      nextKey = key + 1
-    }
-  })
-
-  return nextKey
-}
+import ButtonNoBorder from '../../Atoms/ButtonNoBorder'
+import FlexBox from '../../Templates/FlexBox'
+import { TrashAlt } from 'styled-icons/boxicons-solid/TrashAlt'
 
 const Duplicable = ({
-  model,
-  addType,
-  canBeEmpty = false,
-  instancesProps = [],
+  Model,
+  addText,
+  requireContent = false,
+  instancesCount = requireContent ? 1 : 0,
+  onRemove = () => {},
+  onAdd = () => {},
 }) => {
-  if (instancesProps.length === 0 && !canBeEmpty) {
-    instancesProps.push({})
-  }
-
-  const [duplicated, setDuplicated] = useState(
-    instancesProps.map(
-      (instanceProps, index) => React.cloneElement(model, { ...instanceProps, key: index, index }),
-    ),
-  )
-
-  const nextKey = computeNextKey(duplicated)
+  const [{ lastIndex, indices }, setState] = useState({
+    lastIndex: instancesCount - 1,
+    indices:   [...Array(instancesCount).keys()],
+  })
 
-  const removeDuplicate = key => setDuplicated(
-    duplicated
-      .filter(duplicate => (!canBeEmpty && duplicated.length < 2) || duplicate.key !== key)
-      .map((filtered, index) => ({ ...filtered, index })),
-  )
+  const addDuplicata = () => {
+    setState({
+      lastIndex: lastIndex + 1,
+      indices:   [
+        ...indices,
+        lastIndex + 1,
+      ],
+    })
+    onAdd(lastIndex + 1)
+  }
 
-  return <>
-    {
-      duplicated.map(duplicate => {
-        return React.cloneElement(duplicate, { ...duplicate, onRemove: () => removeDuplicate(duplicate.key) })
+  const removeDuplicata = key => {
+    if (!requireContent || indices.length > 1) {
+      setState({
+        lastIndex: lastIndex,
+        indices:   indices.filter(index => index !== key),
       })
+      onRemove(key)
     }
+  }
+
+  return <>
     {
-      React.cloneElement(addType, {
-        onClick: () => setDuplicated(
-          [
-            ...duplicated,
-            React.cloneElement(model, { key: nextKey, index: duplicated.length }),
-          ],
-        ),
-        ...addType.props,
-      })
+      indices.map(index => <FlexBox alignItems="center" mb={1} gap={1} key={index}>
+        <Model index={index} />
+        {
+          (!requireContent || indices.length > 1) &&
+          <FlexBox
+            component={TrashAlt}
+            size={24}
+            cursor="pointer"
+            onClick={() => removeDuplicata(index)}
+            data-testid="InputRemover"
```

Comment:
that's bad, `data-testid` props should remains in test files

Reply:
fixed in https://github.com/rgsystemes/rgsupv-dashboard/pull/2669/commits/c2a0cec683a8c880f80dbe7ffa1237f714dc9ee2